cherryontech/unmatched
# #UM-12 search bar

## Description

This PR adds the search feature to our site. It allows users to search by name, returning results where any part of the name matches their search. It also allows users to clear search results, so they can once again view all profiles. This PR also adds the 'Outfit' Google font as the default font, as determined by our designers. This PR also lifts up certain states so that they can be passed down more efficiently to several components (specifically the data of the professionals).

## Related Tickets & Documents
Closes #UM-12

## Acceptance Criteria


- [x] A search bar is prominently displayed on the page.
- [x] When I type a keyword and press "Enter" or click the search icon, the page displays profiles that match the keyword.
- [x] The search results display relevant information about the individual, including their name, title, and a short bio.
- [x] If no matches are found, a message is displayed indicating no results.

## What type of PR is this? (check all applicable)


|     | Type                       |
| --- | -------------------------- |
| ✓ | :sparkles: New feature     |
| ✓ | 🎨 Style                   |
|     | :hammer: Refactoring       |
|     | 🔥 Performance Improvements|
|     | :bug: Bug fix              |
|     | :100: Add tests            |
| ✓ | :link: Update dependencies |
|     | :scroll: Docs              |

## Mobile & Desktop Screenshots/Recordings

https://github.com/cherryontech/unmatched/assets/115492619/90ad11bb-9b6c-4680-ab02-d7877fe20778



## What I learned
I learned more about the difference between `useRef` and `useState` and how if the state is never getting set (remains constant) then it might not make sense to store an array, for example, in a state and instead store it in useRef which is mutable without triggering re-renders.

## Testing steps
1. Switch to this branch with the following command: `git switch feature/UM-12-search-bar`
2. Install dependencies with the following command: `npm i`
3. Run the following scripts: `npm run dev` to start Vite's dev build & `npm run tailwind` to start the Tailwind CSS build
4. View the app in browser (the `npm run dev` script will return the url you can see the app at)
5. Navigate to the the Resources page and search for a name (or part of a name) that you see on the page in the search bar. Verify that you see every result where any part of the name matches the string you inputted. 
6. Verify that after you search, a text block appears under the search bar telling you what search results you're seeing. Also verify that you cannot type in the search bar and are prompted to clear the search. Clear the search.
8. Now search for a name/word that isn't part of any of the profile names (e.g. 'tomato'). 
9. Verify that a message appears that there are no matching results. 
10. Read through all the comments added to the code below and interact with them by asking questions and adding comments.

## Added to documentation?

|     | Type                       |
| --- | -------------------------- |
|     | 📜 README.md    |
|  ✓ | 🙅 no documentation needed    |


## What gif best describes this PR or how it makes you feel?

![gif](https://media.giphy.com/media/cnp5B63gSse2c/giphy.gif)
